int32 period
int32 port
int32 subnetwork